File: dev/20190704/k2works/api/rental/domain/movie.rb
require_relative './regular_price'
require_relative './new_release_price'
require_relative './children_price'

module Domain
  class Movie
    REGULAR = 0
    NEW_RELEASE = 1
    CHILDREN = 2

    attr_reader :title
    attr_reader :price_code

    def price_code=(value)
      @price_code = value
      @price = case price_code
               when REGULAR; RegularPrice.new
               when NEW_RELEASE; NewReleasePrice.new
               when CHILDREN; ChildrenPrice.new
               end
    end

    def initialize(title, the_price_code)
      @title, @price_code = title, the_price_code
    end

    def charge(days_rented)
      @price.charge(days_rented)
    end

    def frequent_renter_points(days_rented)
      (price_code == NEW_RELEASE && days_rented > 1) ? 2 : 1
    end

  end
end

File: dev/20190710/k2works/api/video_rental/domain/movie.py
class Movie:
    REGULAR = 0

    NEW_RELEASE = 1

    CHILDREN = 2

    def __init__(self, title, price_code):
        self.__title = title

        self.__price_code = price_code

    @property
    def title(self):
        return self.__title

    @property
    def price_code(self):
        return self.__price_code
